libs/media-player/src/lib/MediaPlayer.tsx

```diff
@@ -21,9 +19,17 @@ export function MediaPlayer({ audioUrl, listenMessage }: MediaPlayerProps) {
         }
 
         audio.play();
+        setIsPlaying(true);
+    };
+
+    const handlePause = () => {
+        audio?.pause();
+        setIsPlaying(false);
```

Comment:
Do not call `setIsPlaying` to manage state. You have introduced a second source of truth for state by doing so. Instead, use event listeners \ useEffect on the actual audio element's state.